chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+        cmd = b"\x52" + address + b"\x20"
+        _rawsend(radio, cmd)
+        data = _recv(radio, address, 32)
+
+        if address != b"\x00\x00":
+            _rawsend(radio, COMMAND_ACCEPT)
+            response = _rawrecv(radio, 1)
+            if response != COMMAND_ACCEPT:
+                raise errors.RadioError("Radio refused to read block at %04s"
+                                        % util.hexprint(address))
+
+        return data
+    except Exception:
+        raise errors.RadioError("Failed to read block from radio at %04s"
+                                % util.hexprint(address))
+
+
+def _write_block(radio, channel_id, data):
+    address = bytes(_get_memory_address(channel_id))
+
+    try:
+        cmd = b"\x57" + address + b"\x10" + data
+        _rawsend(radio, cmd)
+
+        response = _rawrecv(radio, 1)
+        if response != COMMAND_ACCEPT:
+            raise errors.RadioError("Radio refused to write block at %04s"
+                                    % util.hexprint(address))
+    except Exception:
+        raise errors.RadioError("Failed to write block to radio at %04s"
+                                % util.hexprint(address))
+
+
+def _encode_frequency(freq_mhz):
+    """
+    Encodes a frequency in MHz into 4 bytes.
+    Converts from:
+        MHz → Byte3 Byte2 . Byte1 Byte0
+    Example:
+        431.350 → 00 50 13 43
+    """
+    # TODO: make limitation dynamic based on location
+    if freq_mhz == 0.0:
+        return b'\xff\xff\xff\xff'
+
+    freq = int(freq_mhz / 10)
+
+    hex_str = str(freq)
+    if len(hex_str) % 2 != 0:
+        hex_str += "0"
+    while len(hex_str) < 8:
+        hex_str = "00" + hex_str
+    return bytes.fromhex(hex_str)[::-1]
+
+
+def _decode_freq(hex_bytes):
+    """
+    Decodes a 4-byte frequency representation into MHz.
+    The bytes must be read **right to left** in HEX format (LBCD):
+        Byte3 Byte2 . Byte1 Byte0
+    Example:
+        00 50 13 43  ->  431.350 MHz
+    """
+    b0, b1, b2, b3 = hex_bytes
+
+    if b0 == 0xFF and b1 == 0xFF and b2 == 0xFF and b3 == 0xFF:
+        return 0.0
+
+    # Extract the high and low nibbles (hex digits) of b2
+    b2_high = (b2 >> 4) & 0xF  # Get the first hex digit (upper nibble)
+    b2_low = b2 & 0xF          # Get the second hex digit (lower nibble)
+
+    form = f'{b3:02X}{b2_high:X}{b2_low:X}{b1:02X}{b0:02X}'
+    return int(form)
+
+
+def _decode_channel_settings(byte):
+    """
+    Decodes the main channel settings byte into a structured dictionary.
+
+    :param byte: The byte to decode (integer)
+    :return: Dictionary with settings
+    """
+    return {
+        # (1=Wide, 0=Narrow)
+        "W/N Mode": "Wide" if ord(byte) & 0x10 else "Narrow",
+        "Compander": bool(ord(byte) & 0x80),  # (1=ON, 0=OFF)
+        "Scramble Mode": ord(byte) & 0x07,  # (0=OFF, 1-7=Scramble levels)
+        "Special QT/DQT": "A" if ord(byte) & 0x20 else
+        ("B" if ord(byte) & 0x40 else "OFF"),
```

Comment:
Please use bitwise bitfields.. the purpose of that is to avoid this hand-masking which tends to be error-prone and is much harder to read.